extensions/warp-ipfs/src/store/message.rs

```diff
@@ -1778,6 +2796,51 @@ async fn process_conversation(
 
             this.delete_conversation(conversation_id, false).await?;
         }
+        ConversationEvents::NewCommunityInvite {
+            community_id,
+            invite,
+            community_document,
+        } => {
+            let did = this.identity.did_key();
+
+            if this.contains_community(community_id).await {
+                return Err(anyhow::anyhow!("Already apart of {community_id}").into());
+            }
+
+            let recipients = community_document.participants().clone();
+
+            for recipient in &recipients {
+                if !this.discovery.contains(recipient).await {
+                    let _ = this.discovery.insert(recipient).await;
+                }
+            }
+
+            this.set_community_document(community_document).await?;
+
+            this.create_community_task(community_id).await?;
```

Comment:
If we want to initialize the task after receiving an invite, it might be worth having a flag there to make sure we arent subscribed to the main topics until after the invitation is accepted to prevent receiving messages in the background.

Reply:
Good point. I'll implement this in my next PR where I will be implementing messaging.